src/hash_table.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding utf-8 -*-
+"""Implementation of a quick sort."""
+
+WORD_LIST = '/usr/share/dict/words'
+
+
+def add_hash(input):
+    """additive hash"""
+    input = input.encode()
+    hash_value = 0
+    my_array = bytearray(input)
+    for num in my_array:
+        hash_value += num
+    return hash_value
+
+
+def xor_hash(input):
+    """binary xor hash"""
+    input = input.encode()
+    hash_value = 0
+    my_array = bytearray(input)
+    for num in my_array:
+        hash_value = num ^ hash_value
+    return hash_value
+
+
+def rot_hash(input):
+    """rotating hash"""
+    input = input.encode()
+    hash_value = 0
+    my_array = bytearray(input)
+    for num in my_array:
+        hash_value = (hash_value << 4) ^ (hash_value >> 28) ^ num
+    return hash_value
+
+
+def sax_hash(input):
+    """saxiphone hash"""
+    input = input.encode()
```

Comment:
Couple things here. First, don't assign the variable name `input` because that is a builtin python function -- i.e. a reserved variable name.

Secondly, this encode statement won't work on all combinations of Python 2/Python 3/unicode/bytes. You're getting errors in Python 2 with strings containing non-ascii chars. try `encode("utf-8")` so it can encode non-ascii chars. the default encoding of `.encode()` is ascii.